app/src/main/java/it/quip/android/adapter/SearchArrayAdapter.java

```diff
@@ -1,72 +1,99 @@
-            convertView = LayoutInflater.from(getContext())
-                    .inflate(R.layout.item_user, parent, false);
+    public void setOnClickListener(OnClickListener onClickListener) {
+        this.onClickListener = onClickListener;
+    }
 
-            holder = new ViewHolder();
-            holder.tvName = (TextView) convertView.findViewById(R.id.tv_name);
+    @Override
+    public SearchHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        Context context = parent.getContext();
+        LayoutInflater inflater = LayoutInflater.from(context);
+        mSelected = new HashSet<String>();
 
-            convertView.setTag(holder);
-        } else {
-            holder = (ViewHolder) convertView.getTag();
-        }
+        return getViewHolder(inflater, parent);
+    }
 
-        holder.tvName.setText(getSearchName(value));
+    @Override
+    public void onBindViewHolder(final SearchHolder viewHolder, final int position) {
+        T value = mValues.get(position);
+        viewHolder.tvName.setText(getName(value));
+        Picasso.with(viewHolder.context).load(getImageUrl(value)).transform(new CircleTransformation(4, Color.WHITE)).into(viewHolder.ivProfile);
 
-        convertView.setOnLongClickListener(new View.OnLongClickListener() {
+        viewHolder.layout.setOnClickListener(new View.OnClickListener() {
             @Override
-            public boolean onLongClick(View v) {
-                if (onLongClickListener != null) {
-                    return onLongClickListener.onLongClick(position, getItem(position));
+            public void onClick(View v) {
+                if (null != onClickListener) {
+                    T value = mValues.get(position);
+                    toggleChecked(viewHolder, value);
+                    onClickListener.onClick(position, value);
                 }
-
-                return false;
             }
         });
 
-        return convertView;
+        setChecked(viewHolder, value);
     }
 
-    public interface OnLongClickListener <T extends ParseObject> {
-        boolean onLongClick(int position, T value);
+    @Override
+    public int getItemCount() {
+        return mValues.size();
     }
 
-    public class ViewHolder {
-        TextView tvName;
+    private void setChecked(SearchHolder viewHolder, T value) {
+        String uid = value.getObjectId();
+        if (mSelected.contains(uid)) {
+            Picasso.with(viewHolder.context).load(R.drawable.ic_checked).into(viewHolder.ivChecked);
+        } else {
+            Picasso.with(viewHolder.context).load(R.drawable.ic_unchecked).into(viewHolder.ivChecked);
+        }
     }
 
-    protected abstract String getSearchName(T value);
+    private void toggleChecked(SearchHolder viewHolder, T value) {
+        String uid = value.getObjectId();
+        if (mSelected.contains(uid)) {
+            Picasso.with(viewHolder.context).load(R.drawable.ic_unchecked).into(viewHolder.ivChecked);
```

Comment:
Might be overkill to use Picasso for this instead of just using `setImageResource` on the imageview